### ISSUE TITLE
Programmatic example is hard to follow

### ISSUE THREAD

**Issue by johnament** [eclipse/microprofile-config#157]
The following example exists in the config spec:

```
[source, java]
----
public class ConfigUsageSample {

    public void useTheConfig() {
        // get access to the Config instance
        Config config = ConfigProvider.getConfig();

        String serverUrl = config.getValue("acme.myprj.some.url", String.class);
        Integer serverPort = config.getValue("acme.myprj.some.port", Integer.class);

        callToServer(serverUrl, serverPort);
    }
}
----

If you need to access a different server then you can e.g. change the configuration via a `-D` system property:

[source, text]
----
$> java -jar some.jar -Dacme.myprj.some.url=http://other.server/other/endpoint
----

```

The value of `acme.myprj.some.port` is not set.  In addition, since we're passing in a URL, its not clear why we would need a port.

**Comment by struberg** [comment]
Hmm right, the url usually contains any port already.

probably s/url/host/ ? host + port makes more sense, isn't?

**Comment by Emily-Jiang** [comment]
fixed.
